lib/samples/edit_feature_attachments/edit_feature_attachments_sample.dart

```diff
@@ -0,0 +1,384 @@
+                        children: [
+                          IconButton(
+                            icon: const Icon(Icons.remove_red_eye),
+                            onPressed: () => viewAttachment(attachments[index]),
+                          ),
+                          IconButton(
+                            icon: const Icon(Icons.delete),
+                            onPressed: () =>
+                                deleteAttachment(attachments[index]),
+                          ),
+                        ],
+                      ),
+                    );
+                  },
+                ),
+              ]),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+
+  // delete an attachment from the selected feature.
+  void deleteAttachment(Attachment attachment) async {
+    setState(() => isLoading = true);
+
+    await widget.arcGISFeature.deleteAttachment(attachment).then((_) {
+      widget.applyEdits(widget.arcGISFeature);
+    });
+    _loadAttachments();
+
+    setState(() => isLoading = false);
+  }
+
+  // view an attachment from the selected feature in a dialog.
+  void viewAttachment(Attachment attachment) async {
+    setState(() => isLoading = true);
+
+    final data = await attachment.fetchData();
+    setState(() => isLoading = false);
+
+    // display the attachment image/pdf file in a dialog.
+    setState(() {
+      showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(
+            title: Text(attachment.name),
+            content: SizedBox(
+              height: 300,
+              child: Image.memory(data),
+            ),
+            actions: [
+              IconButton(
+                icon: const Icon(Icons.save),
+                tooltip: 'Save',
+                onPressed: () {
+                  // save the attachment to the device.
+                  FilePicker.platform.saveFile(
+                    dialogTitle: 'Save Attachment',
+                    fileName: attachment.name,
+                    bytes: data,
+                    lockParentWindow: true,
+                  );
+                },
+              ),
+              IconButton(
+                icon: const Icon(Icons.close),
+                tooltip: 'Close',
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      );
+    });
+  }
+
+  // add an attachment to the selected feature by FilePicker.
+  void addAttachment() async {
+    var status = false;
+    final result = await FilePicker.platform.pickFiles(
+      allowMultiple: false,
+      type: FileType.custom,
+      onFileLoading: (filePickerStatus) {
+        if (filePickerStatus == FilePickerStatus.done) {
+          status = true;
+        }
+      },
+      allowedExtensions: ['png', 'jpg', 'jpeg', 'pdf', 'txt'],
+    );
+
+    if (result != null && status) {
```

Comment:
When I was testing this on my machine, I had to remove `&& status`. For some reason, I was never getting `status` set to `true`. Is the `status` variable necessary? Is the check for `result != null` good enough by itself?

Reply:
It was used to try fixing the 'resource is not closed issue' before, it is not necessary, and remove it.